r2/r2/controllers/front.py

```diff
@@ -866,7 +850,7 @@ def GET_related(self, num, article, after, reverse, count):
         end = int(time_module.mktime((article._date + rel_range).utctimetuple()))
         nsfw = u"nsfw:0" if not (article.over_18 or article._nsfw.findall(article.title)) else u""
         query = u"(and %s timestamp:%s..%s %s)" % (query, start, end, nsfw)
```

Comment:
The construction of this "related" query probably needs to be different based on the supported syntax(es) of the provider.